### PR TITLE
fix: scalar and nan stacking

### LINKED ISSUE #1085: Document Array of `Image`s cannot be stacked twice when tensor_type=NdArray
**Describe the bug**
Document Array of Images cannot be stacked twice

**To Reproduce**
script.py
```python
from docarray import DocumentArray
from docarray.documents import Image

BATCH_SIZE: int = 1024
da_image: DocumentArray[Image] = DocumentArray[Image]([Image() for _ in range(BATCH_SIZE)])

print("Stacking once")
da_image.stack()
print("Stacking twice")
da_image.stack()
print("Done!")
```

Output
```console
Stacking once
Stacking twice
Traceback (most recent call last):
  File "rabbit.py", line 12, in <module>
    da_image.stack()
  File "/home/jackmin/Documents/da-testing/docarray/docarray/array/array.py", line 204, in stack
    return DocumentArrayStacked.__class_getitem__(self.document_type)(self)
  File "/home/jackmin/Documents/da-testing/docarray/docarray/array/array_stacked.py", line 66, in __init__
    self.from_document_array(docs)
  File "/home/jackmin/Documents/da-testing/docarray/docarray/array/array_stacked.py", line 71, in from_document_array
    self._columns = self._create_columns(docs, tensor_type=self.tensor_type)
  File "/home/jackmin/Documents/da-testing/docarray/docarray/array/array_stacked.py", line 150, in _create_columns
    type_.get_comp_backend().empty(
  File "/home/jackmin/Documents/da-testing/docarray/docarray/computation/numpy_backend.py", line 75, in empty
    raise NotImplementedError('Numpy does not support devices (GPU).')
NotImplementedError: Numpy does not support devices (GPU).
```

**Expected behavior**
```console
Stacking once
Stacking twice
Done!
```

### PR DESCRIPTION
**Pull request by Jackmin801**
Goals:

- Fixes #1085

### CODE REVIEW

File: docarray/typing/tensor/abstract_tensor.py [hunk 1]
```diff
@@ -265,3 +265,9 @@ def _docarray_to_json_compatible(self):
         :return: a representation of the tensor compatible with orjson
         """
         ...
+
+    @property
+    @abc.abstractmethod
+    def ndim(self) -> int:
```
Comment (JoanFM) [inline docarray/typing/tensor/abstract_tensor.py:271]: why is this needed?

Comment (JoanFM) [reply docarray/typing/tensor/abstract_tensor.py:271]: shouldn't the parents already have that?

**Comment by JoanFM** [comment]
Does this issue close any open ticket?

**Review by JoanFM — APPROVED** [review]

**Comment by JoanFM** [comment]
@Jackmin801 do we know if this helped with the `TorchDataset` performance?

**Comment by Jackmin801** [comment]
Nope. It does not improve `TorchDataset` performance on `Image` `DocumentArray`s.
But it does show why stacking `Image` is so expensive. We stack nans uncecessarily.
